### PR TITLE
Add SVG sanitation service

### PR DESCRIPTION
**Pull request by idanya**

### CODE REVIEW

File: src/platform/management/requests/sanitation-request.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { ISource, Source } from '../job/source';
+
+export interface ISanitationRequest {
+  source: ISource;
+  destination: ISource;
+  optimize: boolean;
+  sanitize: boolean;
+}
+export class SanitationRequest {
```
Comment (lelikg) [inline src/platform/management/requests/sanitation-request.ts:9]: Rename to sanitization

Sanitation = garbage department

File: src/platform/management/responses/sanitation-response.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+export interface ISanitationNode {
+  name: string;
+  value: string;
+}
+export interface ISanitationParams {
+  sanitized: boolean;
+  removedNodes: ISanitationNode[];
+}
+export interface ISanitationResult {
```
Comment (lelikg) [inline src/platform/management/responses/sanitation-response.ts:11]: Is this interface needed? You can nest it inside the containing interface

File: src/public/platform/uploader/queued-file-uploader.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ import { UploadJob } from './upload-job';
  * @constructor
  */
 export class QueuedFileUploader {
-  public queue: AsyncQueue<UploadJob>;
+  public queue: QueueObject<UploadJob>;
```
Comment (lelikg) [inline src/public/platform/uploader/queued-file-uploader.ts:13]: Why is this change needed?

File: test/server/platform/management/replies/svg-sanitation-response.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{
+  "code": 0,
+  "message": "OK",
+  "payload": {
+    "sanitationResult": {
```
Comment (lelikg) [inline test/server/platform/management/replies/svg-sanitation-response.json:5]: Are you sure this is not sanitizationResult?

File: test/server/platform/management/sanitation-manager.spec.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+  const apiServer = nock('https://manager.com/').defaultReplyHeaders({
+    'Content-Type': 'application/json',
+  });
+
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  it('perform SVG sanitation', () => {
```
Comment (lelikg) [inline test/server/platform/management/sanitation-manager.spec.ts:25]: You could use async await

**Review by lelikg — CHANGES_REQUESTED** [review]

File: src/platform/management/requests/sanitation-request.ts (hunk 1 above)
Comment (idanya) [reply src/platform/management/requests/sanitation-request.ts:9]: It's a `SanitationManager` (the department) which receives a `SanitationRequest` (the paper) for `Sanitization` (the procedure)

File: src/platform/management/responses/sanitation-response.ts (hunk 2 above)
Comment (idanya) [reply src/platform/management/responses/sanitation-response.ts:11]: Following the Python and JAVA SDK lines. Critical?

File: src/public/platform/uploader/queued-file-uploader.ts (hunk 3 above)
Comment (idanya) [reply src/public/platform/uploader/queued-file-uploader.ts:13]: Linter warning. `AsyncQueue` is deprecated.

File: test/server/platform/management/replies/svg-sanitation-response.json (hunk 4 above)
Comment (idanya) [reply test/server/platform/management/replies/svg-sanitation-response.json:5]: yep. this is how the server is implemented. we'll probably have to go over back and front on those typos :)

File: src/public/platform/uploader/queued-file-uploader.ts (hunk 3 above)
Comment (lelikg) [reply src/public/platform/uploader/queued-file-uploader.ts:13]: but we don't know what effect this will have

**Review by lelikg — APPROVED** [review]
